innersource-license.md

```diff
@@ -0,0 +1,83 @@
+
+## Problem
+
+When two companies within a group want to share code with each other, they need an agreement about the terms and often a legal contract. Creating such agreements on a per project basis takes effort and creates a barrier for sharing. i.e. a team/company might decide not to share their source code with another company in the group because it seems complicated.
+
+Barriers for sharing can lead to silos and duplication of effort in rebuilding similar solutions in multiple parts of the group.
+
+At the time of sharing the source code, it can not be reliably predicted what the value of sharing will be. Hence if the activity of sharing has any significant cost, the companies are unlikely to do it, as they are concerned about the return on investment.
+
+## Context
+
+- A large group with many companies (subsidiaries) that want to share code. (when the group gets larger, the value of this pattern increases)
+- These companies are legally independent units.
+- Multiple of these companies/subsidiaries are developing software, and are using services of the other companies. They have a motivation to contribute to each other’s source code.
+- Certain size of developers (crowdsourcing and ecosystem effects of InnerSource)
+
+## Forces
+
+- **Level of effort** required to write formal agreements, especially if they need to take into account technical, legal, and business perspectives.
+- Enterprises have many **internal regulations,** any agreements have to comply with these regulations, e.g. security, privacy, procurement processes, etc. The volume of regulations can make it difficult to assess whether sharing software between two companies is compliant with these regulations, especially when there is no standard procedure.
+- If any of the companies in the group has a **business model** that depends on proprietary code and accounting of licensing fees within the group
+- **Company culture** that isn’t used to sharing code
+- Freedom over using the software leads to competition, and spread of ownership
+
+## Solutions
+
+Creating an **InnerSource License** customized to the needs of the group in question (and their companies). This license needs to be generic enough to be applied to the most important inter-company relationships.
+
+It is important to write the InnerSource License such that it truly allows for OpenSource-like collaborations across organizational boundaries. Therefore the 4 freedoms of free software should be integrated into the license.
+
+The License is written as a formal legal document, and can be used as part of contracts between the subsidiaries to govern the code sharing agreements.
+
+## Resulting Context
+
+With the InnerSource License, we have a tool to share code between companies within our group.
+
+## Known Instances
```

Comment:
I konw of two other organizations using an MIT-based license. I'll contact them and ask if they would want to provide a paragraph about it here or at least allow us to write about it.

This shouldn't stop this PR from being merged though - we can always add it later.

Reply:
Input from other companies with similar models will certainly be interesting. Just as you said, let's move that out of the context of this PR though.

If you want, you can even file a new issue that says "Harden InnerSource License pattern with experience from x and y".

Will resolve this without action.